hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -4847,5 +4800,20 @@ public String getBucketLocation() throws IOException {
     public Path makeQualified(final Path path) {
       return S3AFileSystem.this.makeQualified(path);
     }
+
```

Comment:
Add to the ListingOperationCallbacks

Reply:
Felt like below three methods are generic for context so I added here such that if required in future it can be used by other operations as well. I think the reason you are saying to add in ListingOperationCallbacks is because these are currently only used in Listing Operations.
I can put in Listing as well.